lib/apiservers/engine/backends/system.go

```diff
@@ -87,5 +198,73 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", s.ProductName)
+	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+}
+
+// Utility functions
+
+// Returns the name of the OS on the appliance vm.  Very likely Photon OS.  This
+// function takes the place of Docker's function that relies on other packages.
+// Since VIC will likely run on either Photon OS or devbox (ubuntu), we can
+// make some assumptions that avoid having to pull in other packages.
+func getOperatingSystem() string {
```

Comment:
This will do for now, but reporting the OS on the appliance is pointless in reality - it tells the user nothing about what they can actually run in containers. Luckily the bootstrap image and appliance are currently on the same version, but we cannot expect that to remain the case, so this will need to become significantly more involved in the future as it'll be reporting what the possible operating systems/variants are.